reflect/reflect.go

```diff
@@ -66,19 +85,29 @@ func generateTypeReflect(t reflect.Type) (attr.Type, error) {
 		return ot, nil
 
 	default:
-		return nil, fmt.Errorf("unknown type encountered: %v", t.Kind())
+		return nil, fmt.Errorf("unknown type encountered at %q: %v", path, t.Kind())
 	}
 }
 
-func generateValue(v any) (attr.Value, diag.Diagnostics) {
-	return generateValueReflect(reflect.ValueOf(v))
+func GenerateValue(v any) (attr.Value, diag.Diagnostics) {
+	return generateValueReflect("", reflect.ValueOf(v))
 }
 
-func generateValueReflect(v reflect.Value) (attr.Value, diag.Diagnostics) {
+var valueInProgress = sets.NewString()
```

Comment:
There are thread safety problems here.  Should we thread this through from the public API surface of this package?